src/test/test_libcephsqlite.cc

```diff
@@ -0,0 +1,422 @@
+            dump_error(__FUNCTION__, __LINE__, ss.str().c_str(), sqlite3_errmsg(db));
+            goto out;
+        }
+        if (sqlite3_bind_text(stmt, 2, col_sha256sum.c_str(), strlen(col_sha256sum.c_str()), NULL) != SQLITE_OK) {
+            std::stringstream ss;
+
+            ss << "error: while attempting to sqlite3_bind_text(col_sha256sum) for row " << row;
+            dump_error(__FUNCTION__, __LINE__, ss.str().c_str(), sqlite3_errmsg(db));
+            goto out;
+        }
+
+        int ret = sqlite3_step(stmt);
+        if (ret != SQLITE_DONE) {
+            std::stringstream ss;
+
+            ss << "error:0x" << std::hex << ret << " while attempting to sqlite3_step() for row " << row;
+            dump_error(__FUNCTION__, __LINE__, ss.str().c_str(), sqlite3_errmsg(db));
+            goto out;
+        }
+        if (sqlite3_reset(stmt) != SQLITE_OK) {
+            std::stringstream ss;
+
+            ss << "error: while resetting for row " << row;
+            dump_error(__FUNCTION__, __LINE__, ss.str().c_str(), sqlite3_errmsg(db));
+            goto out;
+        }
+        ++row;
+    }
+out:
+    sqlite3_finalize(stmt);
+}
+
+long int count_db(sqlite3 *db)
+{
+    long        ret = -1;
+    const char *dml = "SELECT COUNT(*) FROM t1;";
+    sqlite3_stmt *stmt = NULL;
+    const char *unused = NULL;
+
+    if (sqlite3_prepare_v3(db, dml, strlen(dml), 0x0, &stmt, &unused) != SQLITE_OK) {
+        dump_error(__FUNCTION__, __LINE__, "error: when preparing", sqlite3_errmsg(db));
+        goto out;
+    }
+
+    if (sqlite3_step(stmt) != SQLITE_ROW) {
+        std::stringstream ss;
+
+        ss << "error: while stepping";
+        dump_error(__FUNCTION__, __LINE__, ss.str().c_str(), sqlite3_errmsg(db));
+        goto out;
+    }
+    ret = sqlite3_column_int64(stmt, 0);
+out:
+    sqlite3_finalize(stmt);
+    return ret;
+}
+
+void list_db(sqlite3 *db)
+{
+    const char *dml = "SELECT * FROM t1;";
+    sqlite3_stmt *stmt = NULL;
+    const char *unused = NULL;
+
+    if (sqlite3_prepare_v3(db, dml, strlen(dml), 0x0, &stmt, &unused) != SQLITE_OK) {
+        dump_error(__FUNCTION__, __LINE__, "error: when preparing", sqlite3_errmsg(db));
+        goto out;
+    }
+
+    while (sqlite3_step(stmt) == SQLITE_ROW) {
+        std::cout << sqlite3_column_text(stmt, 1) << " " << sqlite3_column_text(stmt, 0) << std::endl;
+    }
+out:
+    sqlite3_finalize(stmt);
+}
+
+void delete_db(sqlite3 *db)
+{
+    const char *dml = "DELETE FROM t1; DROP INDEX t1fname; DROP TABLE t1;";
+    sqlite3_stmt *stmt = NULL;
+    const char *unused = NULL;
+
+    if (sqlite3_prepare_v3(db, dml, strlen(dml), 0x0, &stmt, &unused) != SQLITE_OK) {
+        dump_error(__FUNCTION__, __LINE__, "error: when preparing", sqlite3_errmsg(db));
+        goto out;
+    }
+
+    if (sqlite3_step(stmt) != SQLITE_DONE) {
+        dump_error(__FUNCTION__, __LINE__, "error: while deleting table", sqlite3_errmsg(db));
+    }
```

Comment:
IIRC, what you've done so far only executes the first statement `DELETE from t1;`. You need to iteratively run `sqlite3_prepare_v3` and `sqlite3_step` until `unused == NULL` (IIRC).

Reply:
> IIRC, what you've done so far only executes the first statement `DELETE from t1;`. You need to iteratively run `sqlite3_prepare_v3` and `sqlite3_step` until `unused == NULL` (IIRC).

oh, okay;
I'll address it in a bit.